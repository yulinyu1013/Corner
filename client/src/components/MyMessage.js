import React from 'react';

const MyMessage = ({ message }) => {
  if (
    message.attachments[0]
    && message.attachments[0] !== null
    && message.attachments[0].file !== null
  ) {
    if (typeof message.attachments[0].file === 'object') {
      message.attachments[0].file = URL.createObjectURL(message.attachments[0].file);
      console.log('ttttt');
      console.log(message.attachments[0].file);
      console.log(typeof message.attachments[0].file);
      console.log('ttttt');
    }
    if (message.attachments[0].file.split('.').pop() === 'jpg' || message.attachments[0].file.split('.').pop() === 'jpeg' || message.attachments[0].file.split('.').pop() === 'png') {
      return (
        <div
          className="message-img"
          style={{
            float: 'right', marginRight: '18px', color: 'white', backgroundColor: 'goldenrod',
          }}
        >
          <img
            src={message.attachments[0].file}
            alt="message-attachment"
            className="message-image"
            style={{ float: 'right' }}
          />
        </div>
      );
    }
    if (message.attachments[0].file.split('.').pop() === 'mp4' || message.attachments[0].file.split('.').pop() === 'm4v' || message.attachments[0].file.split('.').pop() === 'avi' || message.attachments[0].file.split('.').pop() === 'mov') {
      return (
        <div
          className="message-vid"
          style={{
            float: 'right', marginRight: '18px', color: 'white',
          }}
        >
          <video
            src={message.attachments[0].file}
            width="100%"
            height="auto"
            controls
          >
            <track kind="captions" />
          </video>
        </div>
      );
    }
    if (message.attachments[0].file.split('.').pop() === 'mp3') {
      return (
        <div
          className="message-aud"
          style={{
            float: 'right', marginRight: '18px', color: 'white',
          }}
        >
          <audio
            src={message.attachments[0].file}
            controls
          >
            <track kind="captions" />
          </audio>
        </div>
      );
    }
  }
  return (
    <div
      className="message"
      style={{
        float: 'right', marginRight: '18px', color: 'white', backgroundColor: 'goldenrod',
      }}
    >
      {message.text}
    </div>
  );
};

export default MyMessage;
