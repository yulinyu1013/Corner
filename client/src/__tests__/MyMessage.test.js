import React from 'react';
import renderer from 'react-test-renderer';
import MyMessage from '../components/MyMessage';
import user from '../images/user.png';

const message = {
  sender: {
    username: 'test',
    avatar: {user}
  },
  attachments: [],
  text: 'test',
}

const messagenotext = {
  sender: {
    username: 'test',
    avatar: {user}
  },
  attachments: [],
  text: '',
}

it('renders TheirMessage correctly when sending only text', () => {
  const tree = renderer.create(<MyMessage message = {message}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders TheirMessage correctly when sending only text', () => {
  const tree = renderer.create(<MyMessage message = {messagenotext}/>).toJSON();
  expect(tree).toMatchSnapshot();
});