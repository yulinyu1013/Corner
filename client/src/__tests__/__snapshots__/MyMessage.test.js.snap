// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TheirMessage correctly when sending only text 1`] = `
<div
  className="message"
  style={
    Object {
      "backgroundColor": "goldenrod",
      "color": "white",
      "float": "right",
      "marginRight": "18px",
    }
  }
>
  test
</div>
`;

exports[`renders TheirMessage correctly when sending only text 2`] = `
<div
  className="message"
  style={
    Object {
      "backgroundColor": "goldenrod",
      "color": "white",
      "float": "right",
      "marginRight": "18px",
    }
  }
>
  
</div>
`;
